gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/util/NumberHelper.java

```diff
@@ -28,10 +28,22 @@
  */
 public final class NumberHelper {
 
+    private static byte asByte(int arg) {
+        if (arg > Byte.MAX_VALUE || arg < Byte.MIN_VALUE)
+            throw new ArithmeticException("byte overflow");
```

Comment:
Nit: this technically doesn't qualify for an ArithmeticException but should probably be an IllegalArgumentException since it is bound checking the incoming argument.

Reply:
`Math.addExact` for int and long throw `ArithmeticException`, I do the same for byte and short

Reply:
Their usage is different as it's not used for argument checking, it applies checking to a result. I'm OK with not changing this due to how NumberHelper is set up as it would add complicated exception handling for little gain.